From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ian Wahbe <ian@wahbe.com>
Date: Mon, 24 Apr 2023 15:00:52 -0700
Subject: [PATCH] fork


diff --git a/newrelic/resource_newrelic_nrql_alert_condition.go b/newrelic/resource_newrelic_nrql_alert_condition.go
index d3cac265..9daeafe9 100644
--- a/newrelic/resource_newrelic_nrql_alert_condition.go
+++ b/newrelic/resource_newrelic_nrql_alert_condition.go
@@ -210,14 +210,13 @@ func resourceNewRelicNrqlAlertCondition() *schema.Resource {
 				},
 			},
 			"term": {
-				Type:          schema.TypeSet,
-				MinItems:      1,
-				MaxItems:      2,
-				Optional:      true,
-				Description:   "A set of terms for this condition. Max 2 terms allowed - at least one 1 critical term and 1 optional warning term.",
-				Elem:          termSchemaDeprecated(),
-				ConflictsWith: []string{"critical", "warning"},
-				Deprecated:    "use `critical` and `warning` attributes instead",
+				Type:        schema.TypeSet,
+				MinItems:    1,
+				MaxItems:    2,
+				Optional:    true,
+				Description: "A set of terms for this condition. Max 2 terms allowed - at least one 1 critical term and 1 optional warning term.",
+				Elem:        termSchemaDeprecated(),
+				Deprecated:  "use `critical` and `warning` attributes instead",
 			},
 			"critical": {
 				Type:          schema.TypeList,
diff --git a/scripts/release.sh b/scripts/release.sh
index ea894a38..225168cf 100755
--- a/scripts/release.sh
+++ b/scripts/release.sh
@@ -8,9 +8,9 @@ COLOR_LIGHT_GREEN='\033[1;32m'
 DEFAULT_BRANCH='main'
 CURRENT_GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
-if [ $CURRENT_GIT_BRANCH != $DEFAULT_BRANCH ]; then
+if [ $CURRENT_GIT_BRANCH != $DEFAULT_BRANCH ] && [ $CURRENT_GIT_BRANCH != 'v2' ]; then
   printf "\n"
-  printf "${COLOR_RED} Error: The release.sh script must be run while on the main branch. \n ${COLOR_NONE}"
+  printf "${COLOR_RED} Error: The release.sh script must be run while on the main or v2 branch. \n ${COLOR_NONE}"
   printf "\n"
 
   exit 1
@@ -73,7 +73,7 @@ ${SPELL_CMD} -source text -w ${CHANGELOG_FILE}
 # Commit CHANGELOG updates
 git add ${CHANGELOG_FILE}
 git commit --no-verify -m "chore(changelog): update CHANGELOG for ${VER_NEXT}"
-git push --no-verify origin HEAD:${DEFAULT_BRANCH}
+git push --no-verify origin HEAD:${CURRENT_GIT_BRANCH}
 
 if [ $? -ne 0 ]; then
   echo "Failed to push branch updates, exiting"
@@ -82,7 +82,7 @@ fi
 
 # Create and push new tag
 git tag ${VER_NEXT}
-git push --no-verify origin HEAD:${DEFAULT_BRANCH} --tags
+git push --no-verify origin HEAD:${CURRENT_GIT_BRANCH} --tags
 
 if [ $? -ne 0 ]; then
   echo "Failed to push tag, exiting"
